webapi/src/main/scala/org/knora/webapi/messages/admin/responder/projectsmessages/ProjectsMessagesADM.scala

```diff
@@ -538,6 +563,20 @@ trait ProjectsADMJsonProtocol extends SprayJsonSupport with DefaultJsonProtocol
 
   import org.knora.webapi.messages.admin.responder.usersmessages.UsersADMJsonProtocol._
 
+  implicit val restrictedViewSizeFormat: JsonFormat[RestrictedViewSize] =
+    new JsonFormat[RestrictedViewSize] {
+      override def read(json: JsValue): RestrictedViewSize = json match {
+        case JsString(str) =>
+          Try(RestrictedViewSize.make(str)) match {
+            case Success(result)    => result.asInstanceOf[RestrictedViewSize]
+            case Failure(exception) => deserializationError(s"Could not parse $str", exception)
+          }
```

Comment:
I wonder how this should work…
`RestrictedViewSize.make` never throws and always returns a `Validation`. IMHO this will always fail at runtime with a `ClassCastException` on the `.asInstanceOf` call.

Where is the `restrictedViewSizeFormat` even used?

Reply:
Using `JsonFormat1` didn't work in long term. This is the only way I found it working also making possible to set `RestrictedViewSize` value object as `sealed abstract`. As for what I understand, if there are no `JsonFormat` instances provided, the only way to get things work is to provide own implementation of `write` and `read` methods. If I comment out `implicit val restrictedViewSizeFormat: JsonFormat[RestrictedViewSize]` implementation, test are failing with following error:
```
/Users/mpro/repos/dsp-api/webapi/src/main/scala/org/knora/webapi/messages/admin/responder/projectsmessages/ProjectsMessagesADM.scala:647:15
could not find implicit value for evidence parameter of type ProjectsADMJsonProtocol.this.JF[dsp.valueobjects.RestrictedViewSize] (Cannot find JsonWriter or JsonFormat type class for dsp.valueobjects.RestrictedViewSize)
    jsonFormat(ProjectRestrictedViewSizeResponseADM, "size")
```